src/Lean/Util/FindExpr.lean

```diff
@@ -53,19 +61,13 @@ unsafe def findUnsafe? (p : Expr → Bool) (e : Expr) : Option Expr :=
 
 end FindImpl
 
+/-- Depth-first explores the subexpressions returning the first subexpression such that the predicate `p` returns true.-/
 @[implementedBy FindImpl.findUnsafe?]
-def find? (p : Expr → Bool) (e : Expr) : Option Expr :=
+partial def find? (p : Expr → Bool) (e : Expr) : Option Expr :=
```

Comment:
Let's mark it as `opaque`, and delete the body. I don't see an application for a `partial` reference implementation. We can't use it for reasoning.